needle.pm

```diff
@@ -52,33 +51,20 @@ sub new {
     }
 
     my $self = {};
-
-    # locate the needle file in one of the lookup directories
-    # - This code initializes $json->{file} so it contains the path relative to the lookup
-    #   directory and re-assigns $jsonfile to contain the absolute path.
-    # - A custom needle repo checkout referenced by the $bmwqemu::vars{NEEDLES_DIR} variable
-    #   is considered to support loading custom needles (e.g. with the openqa-clone-custom-git-refspec
-    #   script). The custom needle repo must be within the current working directory (usually the
-    #   openQA worker's pool directory).
-    # - Otherwise, the needle will be looked up under $bmwqemu::vars{PRJDIR}.
-    my @lookup_dirs;
-    my $needles_dir = $bmwqemu::vars{NEEDLES_DIR};
-    push(@lookup_dirs, $needles_dir) if (defined $needles_dir && index($needles_dir, cwd) == 0);
```

Comment:
this cwd call is what kills us. The whole if is pointless due to:

```
isotovideo:$bmwqemu::vars{NEEDLES_DIR} //= "$bmwqemu::vars{PRODUCTDIR}/needles";
```

As such we should move this cwd there. If NEEDLES_DIR is set but doesn't point to cwd, overwrite it.

Reply:
makes sense, I'll test this tomorrow

Reply:
When I move the check to that assignment one can re-assign that variable later to sneak out from the working directory and load needles from everywhere. (With `cwd` this wasn't possible because the working directory change within autotest is not synced to the backend.)

So I should do that check likely in `needle::init` where we also check that the needle directory exists. Then `cwd` would only be executed once per reload.

Reply:
sounds useful. I was already surprised by that inefficiency when we looked at the code together yesterday :)